cmd/guacone/cmd/vulnerability.go

```diff
@@ -473,22 +571,15 @@ func searchPkgViaHasSBOM(ctx context.Context, gqlclient graphql.Client, searchSt
 		var foundHasSBOMPkg *model.HasSBOMsResponse
 		var err error
 
-		// if the initial depth, check if its a purl or an SBOM URI. Otherwise always search by pkgID
-		if nowNode.depth == 0 {
-			if isPurl {
-				pkgResponse, err := getPkgResponseFromPurl(ctx, gqlclient, now)
-				if err != nil {
-					return nil, nil, fmt.Errorf("getPkgResponseFromPurl - error: %v", err)
-				}
-				foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, model.HasSBOMSpec{Subject: &model.PackageOrArtifactSpec{Package: &model.PkgSpec{Id: &pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id}}})
-				if err != nil {
-					return nil, nil, fmt.Errorf("failed getting hasSBOM via purl: %s with error :%w", now, err)
-				}
-			} else {
-				foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, model.HasSBOMSpec{Uri: &now})
-				if err != nil {
-					return nil, nil, fmt.Errorf("failed getting hasSBOM via URI: %s with error: %w", now, err)
-				}
+		// if the initial depth, check if it's a purl or an SBOM URI. Otherwise, always search by pkgID
+		if nowNode.depth == 0 && primaryCall {
```

Comment:
could you elaborate a bit more on what primaryCall is? Given this is in a loop rather than recursing, would primaryCall be static throughout the function?

Reply:
I have included comments regarding the definition of `primaryCall` in the sections where the functions `SearchForSBOMViaPkg` and `SearchForSBOMViaArtifact` are being called, along with descriptions of the functions.